src/Urgency.ts

```diff
@@ -1,10 +1,19 @@
 import type { Task } from './Task';
 
+/**
```

Comment:
Thank you very much for working on the Urgency feature.

However it has nothing to do with the subject of this one, namely 'feat: add feature and settings updates'

The changes to Urgency need to be in a separate PR, please, so that:
- the description of this PR accurately reflects its contents
- those interested in the Urgency changes can review that change in isolation
- the urgency change appears in the auto-generated release notes

Reply:
Tests were failing so fixing these issues was the only way to get the change in in the first place. If separated it will need to go in first before this change. There is a issue with rounding and the priority testing meaning the approximation was moving. This caused test to fail blocking any commit or push.

Reply:
> Tests were failing so fixing these issues was the only way to get the change in in the first place. 

What platform was that on? does this mean it's worth at some point adding CI builds other platforms?

> If separated it will need to go in first before this change. 

That's a great idea.

> There is a issue with rounding and the priority testing meaning the approximation was moving. This caused test to fail blocking any commit or push.

I don't understand what you mean by "priority testing meaning the approximation was moving" but I'm sure it will make sense when I look at the code.

Reply:
> > Tests were failing so fixing these issues was the only way to get the change in in the first place.
> 
> What platform was that on? does this mean it's worth at some point adding CI builds other platforms?

I just set up a CI run to see if the tests failed on Windows - they passed.

https://github.com/obsidian-tasks-group/obsidian-tasks/runs/7055906510?check_suite_focus=true

They run fine on my Mac, and they pass in the existing Ubuntu CI build.

So @sytone please do log how to reproduce your observation that urgency tests were failing. Thanks in advance...

Reply:
@sytone Might this be the issue: #832 
> Urgency tests fail on machine with US daylight savings time / why are the tests picking up the locale/timezone of the machine they run on?